android/src/main/java/io/sentry/react/RNSentryModule.java

```diff
@@ -364,6 +370,30 @@ public void captureScreenshot(Promise promise) {
         promise.resolve(screenshotsArray);
     }
 
+    private static byte[] takeScreenshotOnUiThread(Activity activity) {
+        CountDownLatch doneSignal = new CountDownLatch(1);
+        final byte[][] bytesWrapper = {{}};
+        final Runnable runTakeScreenshot = () -> {
+            bytesWrapper[0] = takeScreenshot(activity, logger, buildInfo);
+            doneSignal.countDown();
+        };
+
+        if (UiThreadUtil.isOnUiThread()) {
+            runTakeScreenshot.run();
+        } else {
+            UiThreadUtil.runOnUiThread(runTakeScreenshot);
+        }
+
+        try {
+            doneSignal.await(SCREENSHOT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            logger.log(SentryLevel.ERROR, "Screenshot process was interrupted.");
+            return null;
+        }
+
+        return bytesWrapper[0];
```

Comment:
M: we can lazy init the bytesWrapper and return null if there's nothing.

Reply:
I'm not sure what you mean. I can check if the first element of the wrapper length is larger than 0. But I have to create the wrapper before the runnable.

Reply:
What I meant is that you create initing `bytesWrapper = {{}};` but actually you can init the variable wit the return of the `takeScreenshot` and return null otherwise, right now you return an empty `bytesWrapper` even if there's nothing in there, so you are locating something unnecessarily.

Reply:
Also you just need `byte[]` and not `byte[][]`

Reply:
That is on purpose because otherwise I run into `Variable used in lambda expression should be final or effectively final`.

Reply:
Ah gotcha, missed that, thanks, maybe add a comment to explain this?

Reply:
Sure thing, added a comment.